### PR TITLE
[strict templates] programming-exercise.component

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Strict Templates
fixed strict template issues in `programming-exercise.component`

### CODE REVIEW

**Review by nicolasruscher — APPROVED** [review]
Apart from the one new function the changes look good! 👍

File: src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts [hunk 1]
```diff
@@ -161,6 +166,9 @@ export class ProgrammingExerciseComponent extends ExerciseComponent implements O
         modalRef.componentInstance.selectedProgrammingExercises = this.selectedProgrammingExercises;
     }
 
+    castToParticipation(participation: SolutionProgrammingExerciseParticipation | TemplateProgrammingExerciseParticipation): Participation {
```
Comment (imstreetspirit) [inline src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts:169]: guess, we decided not using this kind of casting function during the testing session. other than that, seems nice :)

Comment (SimonEntholzer) [reply src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts]: I removed the cast, now it should be better

**Review by imstreetspirit — APPROVED** [review]
tested on ts1, find nothing. fixes seems good now :)

**Review by ivanchimeno — APPROVED** [review]
lgtm

**Review by iozbeyli — APPROVED** [review]
Tested in testing session

**Review by nicolasruscher — APPROVED** [review]

**Pull request merged**